wodles/aws/aws-s3.py

```diff
@@ -402,10 +447,33 @@ def init_db(self):
         if self.legacy_db_table_name in tables:
             self.migrate_legacy_table()
 
+    def get_columns_number(self, table_name):
+        sql_get_row = "SELECT * FROM {table_name} LIMIT 1;"
+        query_get_row = self.db_connector.execute(sql_get_row.format(table_name=table_name))
+        row = query_get_row.fetchone()
+        return len(row)
+
+    def update_trail_progress_table(self):
+        # rename old trail_progress table to legacy_trail_progress
+        self.db_connector.execute(self.sql_rename_migrate_trail_legacy)
+        # create new trail_progress table
+        self.db_connector.execute(self.sql_create_table.format(table_name='trail_progress'))
+        # copy old table in new table adding bucket_path column
+        for row in self.db_connector.execute(self.sql_select_migrate_trail_progress):
+            # inserts old values on the new table
+            self.db_connector.execute(self.sql_mark_complete.format(table_name=self.db_table_name,
+                                                            bucket_path=self.bucket_path,
+                                                            aws_account_id=row[0],
```

Comment:
instead of using `row[0]`, `row[1]`, etc. Loop over rows using the following:
```python
for aws_account_id, aws_region, log_key, processed_date, created_date in self.db_connector.execute(self.sql_select_migrate_trail_progress):
```
This makes the code much easier to understand 😄 .